### ISSUE TITLE
Commit 2195f25 breaks toolchain configurations

### ISSUE THREAD

**Issue by vladmos** [bazelbuild/bazel#2897]
2195f253c8fed1e398718e63b6e7b4d768dc4e1f

As reported by @juhalindfors:

This commit breaks my previously working toolchain configuration. The previous commit 3e25d20 was still ok:

```
> git checkout 2195f25
M	src/java_tools/buildjar/java/com/google/devtools/build/buildjar/javac/BlazeJavacMain.java
Previous HEAD position was 3e25d20... Cleanup SingleJarActionBuilder
HEAD is now at 2195f25... Update to javac9-r4023

> bazel build //src:bazel
.
INFO: Found 1 target...
Target //src:bazel up-to-date:
  bazel-bin/src/bazel
> cd mytest/
> ../bazel-bin/src/bazel build lib --java_toolchain=:testlib-toolchain
Extracting Bazel installation...
.
INFO: Found 1 target...
ERROR: missing input file '@bazel_tools//third_party/java/jdk/langtools:javac-9-dev-r3297-4.jar'.
ERROR: /home/u/.cache/bazel/_bazel_u/425a650790ab9/external/project/BUILD:31:1: @project//:target: missing input file '@bazel_tools//third_party/java/jdk/langtools:javac-9-dev-r3297-4.jar'.
Target //:lib failed to build
```

Toolchain config:

```
java_toolchain(
    name = "testlib-toolchain",
    bootclasspath = ["@bazel_tools//tools/jdk:bootclasspath"],
    encoding = "iso-8859-1",
    extclasspath = ["@bazel_tools//tools/jdk:extdir"],
    forcibly_disable_header_compilation = 0,
    genclass = ["@bazel_tools//tools/jdk:GenClass_deploy.jar"],
    header_compiler = ["@bazel_tools//tools/jdk:turbine_deploy.jar"],
    ijar = ["@bazel_tools//tools/jdk:ijar"],
    javabuilder = ["@bazel_tools//tools/jdk:JavaBuilder_deploy.jar"],
    javac = ["@bazel_tools//third_party/java/jdk/langtools:javac_jar"],
    javac_supports_workers = 1,
    jvm_opts = [
        "-XX:+TieredCompilation",
        "-XX:TieredStopAtLevel=1",
    ],
    misc = [
        "-XDskipDuplicateBridges=true",
    ],
    singlejar = ["@bazel_tools//tools/jdk:SingleJar_deploy.jar"],
    source_version = "8",
    target_version = "8",

    visibility = ["//visibility:public"]
)
```

The issue still persists on current HEAD.

**Comment by cushon** [comment]
I mailed a fix: https://bazel-review.googlesource.com/c/10530/

**Comment by juhalindfors** [comment]
Applied patch set against current HEAD. Worked for me.

```
> git fetch https://bazel.googlesource.com/bazel refs/changes/30/10530/2 
From https://bazel.googlesource.com/bazel
 * branch            refs/changes/30/10530/2 -> FETCH_HEAD
> git checkout FETCH_HEAD
M	src/java_tools/buildjar/java/com/google/devtools/build/buildjar/javac/BlazeJavacMain.java
Note: checking out 'FETCH_HEAD'.
HEAD is now at 9511ba3... Clean up javac and Error Prone targets
> bazel build //src:bazel
> ../bazel-bin/src/bazel build lib --java_toolchain=:testlib-toolchain
Extracting Bazel installation...
.
INFO: Found 1 target...

```

**Comment by cushon** [comment]
Fixed by 8470be1122825aae8ad0903dd1e1e2a90cce47d2
